src/mainloop.rkt

```diff
@@ -78,7 +79,11 @@
   (debug #:from 'progress #:depth 3 "[1/2] Preparing points")
   ;; If the specification is given, it is used for sampling points
   (timeline-event! 'analyze)
-  (*sampler* (make-sampler (*output-repr*) precondition-prog (or specification prog)))
+  (define symmetry-groups (filter (lambda (group) (> (length group) 1)) (connected-components (or specification prog))))
+  (define preprocess-structs (append preprocess (map symmetry-group symmetry-groups)))
+  (*herbie-preprocess* preprocess-structs)
```

Comment:
Ugh, not thrilled about a new global variable.

Reply:
Is there another way to do it? It seems everything follows this pattern.
If we split `pcontext` into two, I would either need two `pcontext` global variables or one struct containing both of them.

Reply:
At least if it's another pcontext I'd feel a bit better about it.